netaddr.go

```diff
@@ -293,51 +229,38 @@ func (ip IP) IsZero() bool { return ip == IP{} }
 // For the zero value (see IP.IsZero), it returns 0.
 // For IP4-mapped IPv6 addresses, it returns 128.
 func (ip IP) BitLen() uint8 {
-	if ip.IsZero() {
+	switch ip.z {
+	case z0:
 		return 0
-	}
-	if ip.ipImpl.is4() {
+	case z4:
 		return 32
 	}
 	return 128
 }
 
 // Zone returns ip's IPv6 scoped addressing zone, if any.
 func (ip IP) Zone() string {
-	if v6z, ok := ip.ipImpl.(v6AddrZone); ok {
-		return v6z.zone
+	if ip.z == nil {
+		return ""
 	}
-	return ""
+	zoneName, _ := ip.z.Get().(string)
```

Comment:
We shouldn't make `intern.Value.Get` return nil when it has a nil receiver, right?

Reply:
I think I'd rather catch mistakes and explode if you use a nil.